src/platform/term.c

```diff
@@ -465,20 +471,33 @@ static void buffer_plot(int ch, int x, int y, fcolor *fg, fcolor *bg) {
 
     intcolor cube_fg, cube_bg;
 
-    coerce_colorcube(fg, &cube_fg);
-    coerce_colorcube(bg, &cube_bg);
-    if (cube_fg.idx == cube_bg.idx) {
-        // verify that the colors are really the same; otherwise, we'd better force the output apart
-        int naive_distance =
-            (fg->r - bg->r) * (fg->r - bg->r)
-            + (fg->g - bg->g) * (fg->g - bg->g)
-            + (fg->b - bg->b) * (fg->b - bg->b);
-        if (naive_distance > 3) {
-            // very arbitrary cutoff, and an arbitrary fix, very lazy
-            if (cube_bg.r > 0) {cube_bg.r -= 1; cube_bg.idx -= 1; }
-            if (cube_bg.g > 0) {cube_bg.g -= 1; cube_bg.idx -= 6; }
-            if (cube_bg.b > 0) {cube_bg.b -= 1; cube_bg.idx -= 36; }
+    if (colormode == coerce_256) {
+        coerce_colorcube(fg, &cube_fg);
+        coerce_colorcube(bg, &cube_bg);
+        if (cube_fg.idx == cube_bg.idx) {
+            // verify that the colors are really the same; otherwise, we'd better force the output apart
+            int naive_distance =
+                (fg->r - bg->r) * (fg->r - bg->r)
+                + (fg->g - bg->g) * (fg->g - bg->g)
+                + (fg->b - bg->b) * (fg->b - bg->b);
+            if (naive_distance > 3) {
+                // very arbitrary cutoff, and an arbitrary fix, very lazy
+                if (cube_bg.r > 0) {cube_bg.r -= 1; cube_bg.idx -= 1; }
+                if (cube_bg.g > 0) {cube_bg.g -= 1; cube_bg.idx -= 6; }
+                if (cube_bg.b > 0) {cube_bg.b -= 1; cube_bg.idx -= 36; }
+            }
         }
+    } else {
```

Comment:
You seem to be also catching `coerce_16` in this case.  Is that intended?

Reply:
Yup! This function is never called when `colormode == coerce_16`